android/java/org/chromium/chrome/browser/settings/BraveNewsPreferencesTypeAdapter.java

```diff
@@ -0,0 +1,500 @@
+                    setSearchUrl(position, viewHolder);
+                } else if (mBraveNewsPreferencesSearchType
+                                == BraveNewsPreferencesSearchType.NewSource
+                        && position <= getItemCount() - mFeedSearchResultItemList.size()) {
+                    int resultPosition = 0;
+                    if (mChannelList.size() > 0) {
+                        resultPosition = mChannelList.size() + ONE_ITEM_SPACE;
+                    }
+                    resultPosition += mPublisherList.size() + ONE_ITEM_SPACE;
+
+                    setFeedSearchResultItem(position - resultPosition, viewHolder);
+                } else {
+                    setSource(position, viewHolder,
+                            mPublisherList.get(position - getChannelItemsCount() - ONE_ITEM_SPACE));
+                }
+
+            } else if (mBraveNewsPreferencesType.equalsIgnoreCase(
+                               BraveNewsPreferencesType.Channels.toString())) {
+                setChannel(position, viewHolder, mChannelList.get(position));
+            }
+        }
+    }
+
+    private void setSearchUrl(int position, NewsPreferencesViewHolder viewHolder) {
+        viewHolder.imagePublisher.setVisibility(View.GONE);
+        viewHolder.imageChannel.setVisibility(View.GONE);
+
+        if (mBraveNewsPreferencesSearchType == BraveNewsPreferencesSearchType.SearchUrl
+                || mBraveNewsPreferencesSearchType == BraveNewsPreferencesSearchType.GettingFeed) {
+            viewHolder.name.setText(mSearchUrl);
+            viewHolder.subtitle.setVisibility(View.GONE);
+            viewHolder.btnFollow.setVisibility(View.VISIBLE);
+            if (mBraveNewsPreferencesSearchType == BraveNewsPreferencesSearchType.GettingFeed) {
+                displayFollowButton(true, R.string.getting_feeds, viewHolder);
+            } else {
+                displayFollowButton(false, R.string.get_feeds, viewHolder);
+            }
+
+            viewHolder.btnFollow.setOnClickListener(view -> {
+                if (mBraveNewsPreferencesSearchType == BraveNewsPreferencesSearchType.SearchUrl) {
+                    mBraveNewsPreferencesSearchType = BraveNewsPreferencesSearchType.GettingFeed;
+                    notifyItemChanged(position);
+                    mBraveNewsPreferencesListener.findFeeds(mSearchUrl);
+                }
+            });
+        } else if (mBraveNewsPreferencesSearchType == BraveNewsPreferencesSearchType.NotFound) {
+            viewHolder.name.setText(
+                    mContext.getResources().getString(R.string.no_feed_found, mSearchUrl));
+            viewHolder.btnFollow.setVisibility(View.GONE);
+            viewHolder.subtitle.setVisibility(View.GONE);
+        }
+    }
+
+    private void setFeedSearchResultItem(int position, NewsPreferencesViewHolder viewHolder) {
+        FeedSearchResultItem feedSearchResultItem = mFeedSearchResultItemList.get(position);
```

Comment:
maybe a check on `position < mFeedSearchResultItemList.size()` ?

Reply:
Sure. It is good to add check.